src/Source/DoctrineORMQuerySourceIterator.php

```diff
@@ -147,6 +158,32 @@ public function getDateTimeFormat()
         return $this->dateTimeFormat;
     }
 
+    /**
+     * @return string An ISO8601 duration
+     */
+    public function getDuration(\DateInterval $dateInterval)
+    {
+        $datePart = '';
+        foreach (self::DATE_PARTS as $datePartAttribute => $datePartAttributeString) {
+            if ($dateInterval->$datePartAttribute !== 0) {
+                $datePart .= $dateInterval->$datePartAttribute.$datePartAttributeString;
+            }
+        }
+
+        $timePart = '';
+        foreach (self::TIME_PARTS as $timePartAttribute => $timePartAttributeString) {
+            if ($dateInterval->$timePartAttribute !== 0) {
+                $timePart .= $dateInterval->$timePartAttribute.$timePartAttributeString;
+            }
+        }
+
+        if ('' === $datePart && '' === $timePart) {
+            return 'P0Y';
```

Comment:
what does P0Y mean? should that also be a const?

Reply:
This looks fine to me, it is a datetime format by php